From 3ab0707d730d0412ea0c5dd61550ad66e55911ce Mon Sep 17 00:00:00 2001
From: Daniel Charles <daniel.charles@intel.com>
Date: Wed, 29 Aug 2012 16:51:00 -0700
Subject: MUST_REVERT: Use h/w decoder for h.264 first for video playback.

This is to allow h.264 main and high profiles playback. Also
baseline profile will use the h/w decoder.

Considered as a temporary solution to allow video playback.  This
temporary solution maps (copies) decoded raw yuv nv12 frames to
user space to push them to the Surface Flinger for rendering. A
color conversion nv12 to rgb585 is performed on cpu before pushing.

This patch is not meant to go upstream.

For: AXIA-328
Depends-Change-Id: Ic75806a996f1fee3a46a02ddbd2f3e4e619e48c9
Depends-Change-Id: I77f025f9aaabfe8db37b960c04b9e7ee73670e0c
Signed-off-by: Daniel Charles <daniel.charles@intel.com>

Change-Id: I1542fb1491f8d789c714d225b554da0a39f06d5e
Signed-off-by: Daniel Charles <daniel.charles@intel.com>
---
 media/libstagefright/AwesomePlayer.cpp |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/media/libstagefright/AwesomePlayer.cpp b/media/libstagefright/AwesomePlayer.cpp
index bd28118..efcf546 100644
--- a/media/libstagefright/AwesomePlayer.cpp
+++ b/media/libstagefright/AwesomePlayer.cpp
@@ -55,7 +55,7 @@
 
 #include <cutils/properties.h>
 
-#define USE_SURFACE_ALLOC 1
+#define USE_SURFACE_ALLOC 0
 #define FRAME_DROP_FREQ 0
 
 namespace android {
@@ -1490,7 +1490,7 @@ status_t AwesomePlayer::initVideoDecoder(uint32_t flags) {
             mClient.interface(), mVideoTrack->getFormat(),
             false, // createEncoder
             mVideoTrack,
-            NULL, flags, USE_SURFACE_ALLOC ? mNativeWindow : NULL);
+            NULL, flags | OMXCodec::kClientNeedsFramebuffer | OMXCodec::kPreferHardwareCodecs, USE_SURFACE_ALLOC ? mNativeWindow : NULL);
 
     if (mVideoSource != NULL) {
         int64_t durationUs;
-- 
1.7.4.1

